File: scratch/find_buttons.py
import os
import re

button_pattern = re.compile(r'<(a|button)[^>]*class="([^"]*(?:px-\d+|py-\d+|btn|rounded-full)[^"]*)"[^>]*>', re.IGNORECASE)
files_with_buttons = {}

for root, _, files in os.walk('.'):
    if 'node_modules' in root or '.git' in root or 'scratch' in root: continue
    for file in files:
        if file.endswith('.php'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8', errors='ignore') as f:
                content = f.read()
                matches = button_pattern.findall(content)
                if matches:
                    valid_matches = []
                    for tag, cls in matches:
                        # exclude non-button things that just have rounded-full
                        if 'w-2' in cls or 'w-5' in cls or 'h-5' in cls or 'badge' in cls or 'w-12' in cls or 'w-14' in cls or 'w-7' in cls or 'w-9' in cls:
                            continue
                        if 'rounded-full' in cls or 'btn' in cls or ('px-' in cls and 'py-' in cls):
                            valid_matches.append(cls)
                    if valid_matches:
                        files_with_buttons[path] = valid_matches

print(f'Found {len(files_with_buttons)} files with buttons.')
for k in list(files_with_buttons.keys()):
    print(k, "->", len(files_with_buttons[k]), "buttons")

File: scratch/find_buttons.php
<?php
$dir = new RecursiveDirectoryIterator('.');
$ite = new RecursiveIteratorIterator($dir);
$files = new RegexIterator($ite, '/^.+\.php$/i', RecursiveRegexIterator::GET_MATCH);

$button_pattern = '/<(a|button)[^>]*class="([^"]*(?:px-\d+|py-\d+|btn|rounded-full)[^"]*)"[^>]*>/i';
$files_with_buttons = [];
$total_buttons = 0;

foreach($files as $file) {
    $path = $file[0];
    if (strpos($path, 'node_modules') !== false || strpos($path, '.git') !== false || strpos($path, 'scratch') !== false) {
        continue;
    }
    
    $content = file_get_contents($path);
    if (preg_match_all($button_pattern, $content, $matches)) {
        $valid_matches = [];
        foreach ($matches[2] as $cls) {
            if (strpos($cls, 'w-2') !== false || strpos($cls, 'w-5') !== false || strpos($cls, 'h-5') !== false || strpos($cls, 'badge') !== false || strpos($cls, 'w-12') !== false || strpos($cls, 'w-14') !== false || strpos($cls, 'w-7') !== false || strpos($cls, 'w-9') !== false) {
                continue;
            }
            if (strpos($cls, 'rounded-full') !== false || strpos($cls, 'btn') !== false || (strpos($cls, 'px-') !== false && strpos($cls, 'py-') !== false)) {
                $valid_matches[] = $cls;
            }
        }
        if (!empty($valid_matches)) {
            $files_with_buttons[$path] = $valid_matches;
            $total_buttons += count($valid_matches);
        }
    }
}

echo "Found " . count($files_with_buttons) . " files with a total of $total_buttons buttons.\n";
foreach ($files_with_buttons as $k => $v) {
    echo "$k -> " . count($v) . " buttons\n";
    // echo "  " . implode("\n  ", $v) . "\n";
}
?>
